src/modules/drop.ts

```diff
@@ -490,26 +497,28 @@ export class DropModule
     factory.allSnapshots().forEach((s) => {
       merkleInfo[s.merkleRoot] = s.snapshotUri;
     });
+    const encoded = [];
     const { metadata } = await this.getMetadata(false);
     invariant(metadata, "Metadata is not set, this should never happen");
     if (factory.allSnapshots().length === 0 && "merkle" in metadata) {
       metadata["merkle"] = {};
-    } else {
+    } else if (!isMetadataEqual(merkleInfo, metadata["merkle"])) {
```

Comment:
i think this is logic bug.

we should not do `else if` because if the code wants to clear out merkle, then it should be able to.
we need to handle the case (as part of testing)
no merkle -> has merkle -> same merkle -> updates merkle -> no merkle

Reply:
what if we just see if there's a change in `metadata["merkle"]` by comparing it with a copy just before processing? it's just strings so I dont think that would be an issue  - [commit](https://github.com/nftlabs/nftlabs-sdk-ts/pull/153/commits/37e4116b9a86f8ede1fdef1515e4b612a55017f2)

Reply:
yeah that's the idea, compare metadata["merkle"] to see if they're the same or different